Update scripting/lua to include information on building LuaIDE/GridHub (#1001)
o3de/o3de.org: content/docs/user-guide/scripting/lua/_index.md

weight: 100
---

> You can use Lua in **Open 3D Engine** (O3DE) to facilitate quick iteration of your game project. Lua is a powerful, fast, lightweight, embeddable scripting language. When you construct new gameplay and game systems, you can run your changes immediately, without compiling your source code.

Reviewer: ```suggestion
You can use Lua in **Open 3D Engine (O3DE)** to facilitate quick iteration of your game project. Lua is a powerful, fast, lightweight, embeddable scripting language. When you construct new gameplay and game systems, you can run your changes immediately, without compiling your source code.
```

Author: @JwGedit This is another case that I could add to the style guide; I may have received other guidance from editors.  A second example I found in the wild: **Open 3D Engine (O3DE) Asset Editor**.

O3DE uses Lua version {{< versions/lua >}}.

## Building Lua Editor and GridHub

> **Lua Editor** (Lua IDE) and GridHub (O3DE's central connection hub for debugging) must be built for every project that requires them. Include `LuaIDE` and `GridHub` as build targets when you build your project: `--target <project>.GameLauncher Editor LuaIDE GridHub`.  

Reviewer: ```suggestion
**Lua Editor** (Lua IDE) and GridHub, which is O3DE's central connection hub for debugging, must be built for every project that requires them. Include `LuaIDE` and `GridHub` as build targets when you build your project: `--target <project>.GameLauncher Editor LuaIDE GridHub`.  
```

> **Lua Editor** (Lua IDE) and GridHub, which is O3DE's central connection hub for debugging, must be built for every project that requires them. Include `LuaIDE` and `GridHub` as build targets when you build your project: `--target <project>.GameLauncher Editor LuaIDE GridHub`.  

Reviewer 2: Need to qualify this build instruction, because LuaIDE and GridHub come pre-built if you build O3DE using the install layout, or install O3DE from the installer.
```suggestion
If you built O3DE from source, you must explicitly include **Lua Editor** (Lua IDE) and GridHub as build targets when you build your project. GridHub is O3DE's central connection hub for debugging, and is required to debug Lua scripts from the Lua IDE. To include these targets, add `LuaIDE` and `GridHub` to your build target list: `--target <project>.GameLauncher Editor LuaIDE GridHub`. 
```

> For information on accessing class references in Lua Editor, refer to [Debugging with Lua Editor](debugging-tutorial).

Reviewer: ```suggestion
For information about accessing class references in Lua Editor, refer to [Debugging with Lua Editor](debugging-tutorial).
```

### Example

If you need to build Lua Editor and GridHub after your project is built, navigate to your O3DE source directory, for example `c:\o3de`, and run the following command, inserting the path to your project's build directory.

```

> cmake --build c:\<o3de-projects>\<project>\build --target LuaIDE GridHub --config profile -- /m

Reviewer 2: Build directory is configurable, not fixed, so let's put that in angle brackets too.
```suggestion
cmake --build c:\<o3de-projects>\<project>\<build> --target LuaIDE GridHub --config profile -- /m
```

  + For information on O3DE's built-in Lua editor, refer to [Lua Editor](./lua-editor).
  + For information about the O3DE EBus, refer to [Working with the Event Bus (EBus) system](/docs/user-guide/engine/ebus).
  
## Section Topics

| Topic | Description |
| --- | --- |
| [Adding Lua Scripts to Component Entities](add-lua-script) | Use the Lua Script component to add script functionality to your game entities. |

> | [Lua Script Structure](basic-lua-script) | Basic Structure of a Lua Script. |

Reviewer: ```suggestion
| [Lua Script Structure](basic-lua-script) | Basic Structure of a Lua script. |
```

> | [Lua Script Structure](basic-lua-script) | Basic Structure of a Lua Script. |

Reviewer: ```suggestion
| [Lua Script Structure](basic-lua-script) | Basic structure of a Lua script. |
```

| [AWS Client Auth Scripting](\docs\user-guide\gems\reference\aws\aws-client-auth\scripting) | Examples of using the AWS Client Auth Gem with Script Canvas and Lua. |
| [AWS Metrics Scripting](\docs\user-guide\gems\reference\aws\aws-metrics\scripting) | Examples of using Script Canvas or Lua with the AWS Metrics Gem to generate and submit metrics. |
| [Gestures Gem](\docs\user-guide\gems\reference\input\gestures) | The Gestures Gem provides detection for common gesture-based input actions. |
| [Virtual Gamepad Gem](\docs\user-guide\gems\reference\input\virtual-gamepad) | The Virtual Gamepad Gem provides controls that emulate a gamepad on touch screen devices for O3DE projects. |
| [Tweener Lua Script](\docs\user-guide\interactivity\user-interface\animating\tweener-system\tweener-lua-code) | Learn how to use Lua scripting to animate your entities with the Scripted Entity Tweener system. |
| [Tweener Timeline](\docs\user-guide\interactivity\user-interface\animating\tweener-system\tweener-timeline) | Use Scripted Entity Tweener's timeline abilities to chain animations together and exert fine control over them. |

> | [Synchronizing Animation Graphs](\docs\user-guide\visualization\animation\character-editor\sync-graph) | Use synchronized animation graphs to synchronize animation between actors. |

Reviewer 2: Whoops.. these links must all use forward slash `/` in their path, not `\`.

Author: Thanks!
